### PR TITLE
add `setTiming` permission

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR adds a `setTiming` permission.

Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

**Comment by mrmr1993** [comment]
!ci-build-me

File: src/lib/transaction_snark/test/util.ml [hunk 1]
```diff
@@ -344,6 +344,10 @@ let permissions_from_update (update : Account_update.Update.t) ~auth =
       ( if Zkapp_basic.Set_or_keep.is_keep update.voting_for then
         default.set_voting_for
       else auth )
+  ; set_timing =
+      ( if Zkapp_basic.Set_or_keep.is_keep update.timing then
+        default.set_voting_for
```
Comment (mitschabaude) [inline src/lib/transaction_snark/test/util.ml:349]: it would probably be better to use `default.set_timing` instead of `default.set_voting_for`

**Review by mitschabaude — APPROVED** [review]
approving with a small comment!

File: src/lib/transaction_snark/test/util.ml (hunk 1 above)
Comment (mrmr1993) [reply src/lib/transaction_snark/test/util.ml:349]: Woops

**Comment by mrmr1993** [comment]
!ci-build-me

File: src/lib/transaction_snark/test/account_timing/account_timing.ml [hunk 2]
```diff
@@ -2045,8 +2045,7 @@ let%test_module "account timing check" =
                 ledger_init_state ;
               Transaction_snark_tests.Util.check_zkapp_command_with_merges_exn
                 ~expected_failure:
-                  Transaction_status.Failure
-                  .Update_not_permitted_timing_existing_account ledger
+                  Transaction_status.Failure.Update_not_permitted_timing ledger
```
Comment (deepthiskumar) [inline src/lib/transaction_snark/test/account_timing/account_timing.ml:2048]: Can we include the update tests involving all possible permissions for sanity? I think including the following should work
```
module Test_input : Transaction_snark_tests.Test_zkapp_update.Input_intf =
struct
  let test_description = "timing"

  let failure_expected =
    Mina_base.Transaction_status.Failure. Update_not_permitted_timing

  let zkapp_update : Account_update.Update.t =
   let timing =
               Zkapp_basic.Set_or_keep.Set
                 ( { initial_minimum_balance =
                       Currency.Balance.of_nanomina_int_exn min_balance
                   ; cliff_time = Mina_numbers.Global_slot.of_int 1000
                   ; cliff_amount =
                       Currency.Amount.of_nanomina_int_exn 100_000_000
                   ; vesting_period = Mina_numbers.Global_slot.of_int 10
                   ; vesting_increment =
                       Currency.Amount.of_nanomina_int_exn 100_000_000
                   }
                   : Account_update.Update.Timing_info.value )
             in
             { Account_update.Update.dummy with timing } 
end

let%test_module "Update account timing" =
  ( module struct
    include Transaction_snark_tests.Test_zkapp_update.Make (Test_input)
  end )
```

**Review by deepthiskumar — APPROVED** [review]
Requested one more test but otherwise looks good!

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Pull request merged**